packages/esm-service-queues-app/src/queue-table/transitions/transfer-queue-dialog.component.tsx

```diff
@@ -0,0 +1,186 @@
+  queueEntry: QueueEntry;
+  closeModal: () => void;
+}
+
+interface FormState {
+  selectedQueue: string;
+  selectedPriority: string;
+  selectedStatus: string;
+}
+
+const TransferQueueDialog: React.FC<ChangeStatusDialogProps> = ({ queueEntry, closeModal }) => {
+  const { t } = useTranslation();
+  const { mutateQueueEntries } = useMutateQueueEntries();
+
+  const [formState, setFormState] = useState<FormState>({
+    selectedQueue: queueEntry.queue.uuid,
+    selectedPriority: queueEntry.priority.uuid,
+    selectedStatus: queueEntry.status.uuid,
+  });
+
+  const { queues } = useQueues();
+
+  const selectedQueue = queues.find((q) => q.uuid == formState.selectedQueue);
+
+  const statuses = selectedQueue?.allowedStatuses;
+  const hasNoStatusesConfigured = selectedQueue && statuses.length == 0;
+  const priorities = selectedQueue?.allowedPriorities;
+  const hasNoPrioritiesConfigured = selectedQueue && priorities.length == 0;
+
+  const setSelectedQueueUuid = (selectedQueueUuid: string) => {
+    const newSelectedQueue = queues.find((q) => q.uuid == selectedQueueUuid);
+    const { allowedStatuses, allowedPriorities } = newSelectedQueue;
+    const newQueueHasCurrentStatus = allowedStatuses.find((s) => s.uuid == formState.selectedStatus);
+    const newQueueHasCurrentPriority = allowedPriorities.find((s) => s.uuid == formState.selectedPriority);
+    setFormState({
+      selectedQueue: selectedQueueUuid,
+      selectedStatus: newQueueHasCurrentStatus ? formState.selectedStatus : allowedStatuses[0]?.uuid,
+      selectedPriority: newQueueHasCurrentPriority ? formState.selectedPriority : allowedPriorities[0]?.uuid,
+    });
+  };
+
+  const setSelectedPriorityUuid = (selectedPriorityUuid: string) => {
+    setFormState({ ...formState, selectedPriority: selectedPriorityUuid });
+  };
+
+  const setSelectedStatusUuid = (selectedStatusUuid: string) => {
+    setFormState({ ...formState, selectedStatus: selectedStatusUuid });
+  };
+
+  const transferQueue = (e) => {
+    e.preventDefault();
+
+    transferQueueEntry(queueEntry.uuid, formState.selectedQueue, formState.selectedStatus, formState.selectedPriority)
+      .then(({ status }) => {
+        if (status === 200) {
+          showSnackbar({
+            isLowContrast: true,
+            title: t('queueEntryTransferred', 'Queue entry transferred'),
+            kind: 'success',
+            subtitle: t('queueEntryTransferredSuccessfully', 'Queue entry transferred successfully'),
+          });
+          mutateQueueEntries();
+          closeModal();
+        } else {
+          throw { message: t('unexpectedServerResponse', 'Unexpected Server Response') };
+        }
+      })
+      .catch((error) => {
+        showSnackbar({
+          title: t('queueEntryTranferFailed', 'Error transferring queue entry'),
+          kind: 'error',
+          subtitle: error?.message,
+        });
+      });
+  };
+
+  return (
+    <div>
+      <Form onSubmit={transferQueue}>
```

Comment:
```suggestion
      <Form onSubmit={transferQueue}>
      	<Stack gap={4}>
```

I recommend using a [Stack](https://react.carbondesignsystem.com/?path=/docs/layout-stack--overview) to space out the form fields in all of these modals.